src/ucp/rndv/rndv_put.c

```diff
@@ -331,10 +331,17 @@ ucp_proto_rndv_put_zcopy_send_progress(uct_pending_req_t *uct_req)
 {
     ucp_request_t *req = ucs_container_of(uct_req, ucp_request_t, send.uct);
     const ucp_proto_rndv_put_priv_t *rpriv = req->send.proto_config->priv;
+    unsigned uct_mem_flags;
+
+    if (req->flags & UCP_REQUEST_FLAG_RNDV_FRAG) {
+        uct_mem_flags = UCT_MD_MEM_ACCESS_LOCAL_READ;
+    } else {
+        uct_mem_flags = 0;
+    }
```

Comment:
can we initiliaze uct_mem_flags in priv struct during protocol initialization? to avoid branch on datapath

Reply:
in the code above seems only UCP_REQUEST_FLAG_RNDV_FRAG is checked.. anyway let's try to avoid this branch in datapath somehow